### PR TITLE
Better inputs design fix placement

### PR DESCRIPTION
**Pull request by aviv1620**
comment on issue #39

### CODE REVIEW

**Comment by aviv1620** [comment]
The Playwright Tests run when the server is busy.
if you run the test again(without change) it works.
recommend increasing the timeout.

**Review by ArkadiK94 — APPROVED** [review]
The layout of the input fields looks great now, thanks. I have only some minor comments for you.

File: src/pages/components/GridSelectorAndLabel.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import { Label } from 'src/pages/components/Label'
+import Grid from '@mui/material/Unstable_Grid2'
+
+export function GridSelectorAndLabel(props: { label: string; children: React.ReactNode }) {
+  return (
+    <>
+      <Grid xs={4}>
```
Comment (ArkadiK94) [inline src/pages/components/GridSelectorAndLabel.tsx:8]: You should make the "xs" configurable so you could use this component in other places (see comments in "RealtimeMapPage")

File: src/pages/RealtimeMapPage.tsx [hunk 2]
```diff
@@ -138,34 +142,46 @@ export default function RealtimeMapPage() {
...
-          </label>
-        </div>
-        <div className="map-header-buttons">
+    <PageContainer className="map-container">
+      <Grid container spacing={2} sx={{ maxWidth: INPUT_SIZE }}>
+        <Grid xs={12}>
+          <Label text={TEXTS.realtime_map_explanation} />
+        </Grid>
+        {/* from date */}
+        <Grid xs={2}>
```
Comment (ArkadiK94) [inline src/pages/RealtimeMapPage.tsx:151]: You should use here the "GridSelectorAndLabel" component

File: src/pages/RealtimeMapPage.tsx [hunk 3]
```diff
@@ -138,34 +142,46 @@ export default function RealtimeMapPage() {
...
+            setTimestamp={(ts) => {
+              const value = ts ? ts.format() : ''
+              setFrom(value)
+              setTo(formatTime(+new Date(value) + (+new Date(to) - +new Date(from)))) // keep the same time difference
+            }}
+            showTimePicker={true}
+          />
+        </Grid>
+        {/* watch locations in range */}
+        <Grid xs={5}>
```
Comment (ArkadiK94) [inline src/pages/RealtimeMapPage.tsx:166]: You should use here the "GridSelectorAndLabel" component

File: src/pages/components/DataAndTimeSelector.tsx [hunk 4]
```diff
@@ -23,27 +22,34 @@ export function DataAndTimeSelector({
       setTimestamp(timeSelected)
   }, [timeSelected])
 
+  useEffect(() => {
+    if (timeSelected != null && !timeSelected.isSame(timestamp)) setTimeSelected(timestamp)
+  }, [timestamp])
+
+  const width = showTimePicker ? '50%' : '100%'
```
Comment (ArkadiK94) [inline src/pages/components/DataAndTimeSelector.tsx:29]: You use a built in word as a variable , it is confusing , please change it.

File: src/pages/components/DataAndTimeSelector.tsx [hunk 5]
```diff
@@ -23,27 +22,34 @@ export function DataAndTimeSelector({
...
+        sx={{ width }}
         value={timeSelected}
         onChange={(ts) => setTimeSelected(ts)}
         format="DD/MM/YYYY"
-        label={TEXTS.choose_datetime}
+        label={TEXTS.choose_date}
         disableFuture
       />
       {showTimePicker && (
         <TimePicker
```
Comment (ArkadiK94) [inline src/pages/components/DataAndTimeSelector.tsx:42]: There is no space between those input fields . Please, add some space.
image

File: src/pages/components/Label.tsx [hunk 6]
```diff
@@ -2,7 +2,11 @@ import React from 'react'
 import styled from 'styled-components'
 
 const StyledDiv = styled.div`
-  min-width: 80px;
+  display: inline-flex;
+  width: 100%;
+  height: 100%;
+  justify-content: center;
```
Comment (ArkadiK94) [inline src/pages/components/Label.tsx:8]: The labels are centered, better to align them to right by "justify-content: flex-start;"
before
image
after
image

File: src/pages/components/StopSelector.tsx [hunk 7]
```diff
@@ -20,7 +19,7 @@ const StopSelector = ({ stops, stopKey, setStopKey }: StopSelectorProps) => {
       value={value}
       onChange={(e, value) => setStopKey(value ? value.key : '0')}
       id="stop-select"
-      sx={{ width: INPUT_SIZE * 2 }}
+      sx={{ width: '100%' }}
```
Comment (ArkadiK94) [inline src/pages/components/StopSelector.tsx:22]: The "StopSelector" component uses input field the same as "RouteSelector" and "OperatorSelector" so you don't need here to set the width

**Review by ArkadiK94 — APPROVED** [review]
The comments :)

File: src/pages/RealtimeMapPage.tsx (hunk 2 above)
Comment (aviv1620) [reply src/pages/RealtimeMapPage.tsx:132]: The layout here needs to be a little bit different from "GridSelectorAndLabel".
The 'xs' for the label need be 2 and the 'xs' for the content need be 10.
To fix it I have two methods.

method 1:
move xsLabel and xsContent  by property.
method 2:
not use  "GridSelectorAndLabel".

Which method to choose?

**Comment by aviv1620** [comment]
OK.
On Tuesday I upstream from main(to prevent conflicts) and then improve the code by the comments.

File: src/pages/RealtimeMapPage.tsx (hunk 2 above)
Comment (ArkadiK94) [reply src/pages/RealtimeMapPage.tsx:132]: In my opinion, the first method is better so you could use this component in different places as well. Let's hear from others to see what they think about it :) @NoamGaash @ShayAdler

File: src/pages/components/DataAndTimeSelector.tsx (hunk 4 above)
Comment (aviv1620) [reply src/pages/components/DataAndTimeSelector.tsx:29]: change the `timestamp` name to `timeValid`.

File: src/pages/components/DataAndTimeSelector.tsx (hunk 5 above)
Comment (aviv1620) [reply src/pages/components/DataAndTimeSelector.tsx]: I refactor the DateAndTimeSelector to two different components and fix the problem.

File: src/pages/components/Label.tsx (hunk 6 above)
Comment (aviv1620) [reply src/pages/components/Label.tsx:8]: done. thanks.

File: src/pages/RealtimeMapPage.tsx (hunk 2 above)
Comment (aviv1620) [reply src/pages/RealtimeMapPage.tsx:135]: I remove the `GridSelectorAndLabel` completely, 
Grid containers and items better to be in the same component.

File: src/pages/components/DataAndTimeSelector.tsx (hunk 4 above)
Comment (ArkadiK94) [reply src/pages/components/DataAndTimeSelector.tsx]: I meant the 'width' variable name, I should have wrote it precisely

Comment (aviv1620) [reply src/pages/components/DataAndTimeSelector.tsx]: OK.
fix in commit [fix space between date and time components](https://github.com/hasadna/open-bus-map-search/pull/61/commits/71f269ef038fe44054d4986684c62ea384e33890) anyway.

the variable not existent anymore.

**Comment by aviv1620** [comment]
fix the comments. 
every page works(test manual).

for another PR:
Playwright has a problem with the `har` file. If I route the traffic to network the test pass.

**Review by ArkadiK94 — COMMENTED** [review]
Thanks, looks good, I have only some comments.

**Comment by ArkadiK94** [comment]
@NoamGaash I reviewed the code , would you like to review it as well?

File: src/pages/RealtimeMapPage.tsx [hunk 8]
```diff
@@ -119,34 +126,54 @@ export default function RealtimeMapPage() {
...
+          <Label text={TEXTS.from_date} />
+        </Grid>
+        <Grid xs={5}>
+          <DataSelector
+            timeValid={moment(from.slice(0, 16))} // remove timezone and seconds
+            setTimeValid={(ts) => {
+              const value = ts ? ts.format() : ''
+              setFrom(value)
+              setTo(formatTime(+new Date(value) + (+new Date(to) - +new Date(from)))) // keep the same time difference
+            }}
```
[the remark this answers is not in the document]
Comment (aviv1620) [reply src/pages/RealtimeMapPage.tsx:145]: the state `form` and the state `to` design before Noam adds the load date from the URL feature.
I prefer to remove the state `to` completely. 
add the minutes when we call `useVehicleLocations`.
and use the `moment.js` library.

this refactor related to issue #79 and issue #71.
so I prefer to solve this in another PR.

**Comment by NoamGaash** [comment]
@ArkadiK94 thank you!
Let's merge it :smile:

**Comment by NoamGaash** [comment]
@all-contributors
please add @aviv1620 for code
please add @ArkadiK94  for review

**Pull request merged**